enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,677 @@
+  provides zone names with the type `local-zone` on the `platform.aws.zones` or
+  `compute[?name=="edge"].platform.aws.zones`.
+
+Phases 1 and 2 are covered by this enhancement.
+
+## Motivation
+
+Presently, the installer does not support Local Zone subnets on day zero operations
+for those reasons:
+- the Local Zone does not support network load balancers (NLB or CLB), used by
+  ingress deployed on worker pool.
+- the Local Zone does not support Nat Gateway, used as a gateway on the default IPI
+  deployment for each subnet and availability zone.
+- the default instance type for compute nodes used on IPI, `m6i.xlarge`, is not
+  available on Local Zones
+- the default EBS type for IPI, `gp3`, is not available in the most Local Zone locations
+- segment the `worker` compute node pool is not allowed
+
+The resources on Local Zones were designed for applications delivering low latency to
+the end-users, in general, hosted in the data plane nodes. Furthermore, the price of
+cloud services is more expensive, and the services are limited. Finally, mixing the
+ingress router backends from different locations in the Local Zones can impact
+drastically, and directly, the end users' experience (ex: routing requests from users
+from NYC to node hosted in Local Zones in Miami, instead of New Year node).
+
+As the applications using edge locations are very specific workloads, cluster admins
+want to have available the compute nodes on the board without mixing cluster components
+on those pools, and would like to specify which workloads will run for each location to
+optimize the cloud resources.
+
+### User Stories
+
+Phase 1:
+
+- As a cluster admin, I want the ability to specify a set of subnets on the AWS
+  Local Zone locations to deploy worker nodes, so I can create custom applications
+  to deliver low latency to my end users.
+
+- As a cluster admin, I would like to create a cluster extending worker nodes to the
+  edge of the AWS cloud provider with Local Zones, so I can create custom applications
+  to deliver low latency to my end users.
+
+- As a cluster admin, I would like to control the workloads running on the edge
+  compute pool, and keep it dedicated for applications which are latency sensitive,
+  so I can have a cost-effective infrastructure, avoiding mixing workloads in different locations.
+
+- As a cluster admin, I would like to select existing subnets from the local and
+  the parent region zones, to install a cluster, so I can manage my custom networks with
+  my automation.
+
+Phase 2:
+
+- As a cluster admin, I would like to install OpenShift clusters, extending the
+  compute nodes to the Local Zones in my day zero operations without needing to
+  set up the network and compute dependencies, so I can speed up the edge adoption
+  in my organization using OKD/OCP.
+
+### Goals
+
+Phase 1:
+
+- Admins can create compute pool named `edge` on the AWS platform to setting up Local Zone MachineSets.
```

Comment:
This wording is a bit confusing to me. Are you talking about specifying a machine pool in the install config or having users pre-provision infrastructure? Not sure what exactly is meant by "compute pool."

Reply:
Thanks for your feedback, @patrickdillon . 
> specifying a machine pool in the install config

This. The goal is to allow to installer allow to create an extra pool on install config named `edge` when installing in existing VPC with Local Zone subnets. The installer will automate the creation of MachineSets for each subnet's zone located on Local Zones.

I would re-phrase it to something like this:
- Admins can create machine pools named `edge` when installing a cluster in existing VPC with subnets in AWS Local Zones.

what do you think?

Reply:
Item [reviewed](https://github.com/openshift/enhancements/pull/1232/commits/fdbdba4c38f3b17c5159b4f5206a6a3671bc74ce#diff-f2590494c9138551ab5b3d5632d11170fe8742a128256f9e1d8ebc63baa19f41R131).